### PR TITLE
#8 - added Download All Feature

### PR DESCRIPTION
**Pull request by gurungrahul2**
added download all button in saved tab to download all posts with one click

### CODE REVIEW

**Comment by aashutoshrathi** [comment]
@gurungrahul2 It works great, but what if we could better zip it using JS and give the user a Single JS file?

**Comment by gurungrahul2** [comment]
> @gurungrahul2 It works great, but what if we could better zip it using JS and give the user a Single JS file?

okay, I will send a follow up commit regarding that.

**Comment by aashutoshrathi** [comment]
@gurungrahul2 Check this.

**Comment by gurungrahul2** [comment]
@aashutoshrathi done. but speed is not that fast but still great. I would suggest you to have a look at console.log too ,as I made it easy to see how many pics downloaded. :)

File: download.js [hunk 1]
```diff
@@ -289,6 +304,63 @@ function injectButtons () {
 function refresh () {
   // as we only can have 8 articles at a time.
   injectButtons()
+  downloadAlltrigger = 1
```
Comment (aashutoshrathi) [inline download.js]: Use boolean instead of int.

Comment (gurungrahul2) [reply download.js]: ok

**Comment by gurungrahul2** [comment]
@aashutoshrathi  ok image1 , image2, image3.... format can also be easily implemented.

**Comment by gurungrahul2** [comment]
@aashutoshrathi showing progress like on button ?

File: download.js [hunk 2]
```diff
@@ -289,6 +304,63 @@ function injectButtons () {
...
+    }
+    if (downloadAlltrigger === true) {
+      console.log('%c Getting Images', 'background: #222; color: #bada55')
+      for (i = 0; i < images.length; i += 1) {
+        var link = images[i].src
+        urls.push(link)
+      }
+      console.log('%c Getting Images', 'background: #222; color: #bada55')
+      urls.forEach(function (url, index) {
+        var filename = 'Image' + index + '.jpg'
```
Comment (aashutoshrathi) [inline download.js]: I think we should use `alt-text` instead.

Comment (gurungrahul2) [reply download.js]: @aashutoshrathi i don't got you.

Comment (gurungrahul2) [reply download.js]: isn't image0, image1, image2 a alt-text ?

Comment (aashutoshrathi) [reply download.js]: No, caption is alt-text

Comment (aashutoshrathi) [reply download.js]: And yeah something like progress button.

Comment (aashutoshrathi) [reply download.js]: Although things are pretty quick

**Comment by aashutoshrathi** [comment]
Does it also takes the videos?

**Comment by gurungrahul2** [comment]
> Does it also takes the videos?

not now.

File: download.js (hunk 2 above)
Comment (gurungrahul2) [reply download.js]: @aashutoshrathi ok lets try

**Comment by gurungrahul2** [comment]
@aashutoshrathi currently JSZip doesn't supports big files, I tried and it only downloads the screenshot of video.

**Comment by gurungrahul2** [comment]
@aashutoshrathi alt-text doesn't contain a caption, it has something like predictions like this image may contain 1 person, this image may contain 2 people etc. I think imageN format is more simple and beautiful.

**Comment by gurungrahul2** [comment]
:)

**Comment by gurungrahul2** [comment]
@aashutoshrathi merging plans ?

**Pull request merged**